components/base/sensorcontrolled/spin.go

```diff
@@ -27,11 +30,30 @@ func (sb *sensorBase) Spin(ctx context.Context, angleDeg, degsPerSec float64, ex
 	// If an orientation movement sensor or controls are not configured, we cannot use this Spin method.
 	// Instead we need to use the Spin method of the base that the sensorBase wraps.
 	// If there is no valid velocity sensor, there won't be a controlLoopConfig.
-	if len(sb.controlLoopConfig.Blocks) == 0 {
+	if sb.controlLoopConfig == nil {
 		sb.logger.CWarnf(ctx, "control parameters not configured, using %v's Spin method", sb.controlledBase.Name().ShortName())
 		return sb.controlledBase.Spin(ctx, angleDeg, degsPerSec, extra)
 	}
 
+	// if loop has been tuned but the values haven't been added to the config, error with tuned values
+	switch {
+	case (sb.configPIDVals[0].NeedsAutoTuning() && !(*sb.tunedVals)[0].NeedsAutoTuning()) &&
+		(sb.configPIDVals[1].NeedsAutoTuning() && !(*sb.tunedVals)[1].NeedsAutoTuning()):
+		err := "\"control_parameters\": ["
+		err += control.TunedPIDErr((*sb.tunedVals)[0]) + ", "
+		err += control.TunedPIDErr((*sb.tunedVals)[1]) + "]"
+		return fmt.Errorf("%v has been tuned, copy these control parameters into the config to enable movement: %v",
+			sb.Name().ShortName(), err)
+	case sb.configPIDVals[0].NeedsAutoTuning() && !(*sb.tunedVals)[0].NeedsAutoTuning():
+		err := control.TunedPIDErr((*sb.tunedVals)[0])
+		return fmt.Errorf("%v has been tuned, copy these linear control parameters into the config to enable movement: %v",
+			sb.Name().ShortName(), err)
+	case sb.configPIDVals[1].NeedsAutoTuning() && !(*sb.tunedVals)[1].NeedsAutoTuning():
+		err := control.TunedPIDErr((*sb.tunedVals)[1])
+		return fmt.Errorf("%v has been tuned, copy these angular control parameters into the config to enable movement: %v",
+			sb.Name().ShortName(), err)
+	default:
+	}
```

Comment:
heres how this code would look with my recommended change. 
```suggestion
if err := control.CheckTuning(sb.Name().Name, sb.configPIDVals, *sb.tunedVals); err != nil {
		return err
	}
	
```